build-tools/packages/build-tools/src/type-test-generator/main.ts

```diff
@@ -12,12 +12,65 @@ import { PackageJson } from "../common/npmPackage";
 import { buildTestCase, TestCaseTypeData } from "../typeValidator/testGeneration";
 import { getFullTypeName, getNodeTypeData, TypeData } from "../typeValidator/typeData";
 import { typeOnly } from "./compatibility";
+import { ExtractorConfig } from "@microsoft/api-extractor";
 
 // Do not check that file exists before opening:
 // Doing so is a time of use vs time of check issue so opening the file could fail anyway.
 // Do not catch error from opening file since the default behavior is fine (exits process with error showing useful message)
 const packageObject: PackageJson = readJsonSync("package.json");
+const previousPackageName = `${packageObject.name}-previous`;
+
+const previousBasePath = `./node_modules/${previousPackageName}`;
+
+const previousPackageJsonPath = `${previousBasePath}/package.json`;
+
+if (!existsSync(previousPackageJsonPath)) {
+	throw new Error(
+		`${previousPackageJsonPath} not found. You may need to install the package via pnpm install.`,
+	);
+}
+const typeRollupFilePath = ExtractorConfig.tryLoadForFolder({
+	startingFolder: previousBasePath,
+});
 
+let typeDefinitionFilePath;
+
+if (typeRollupFilePath) {
+	// Check if a specified typeRollupFile exists
+	if (typeRollupFilePath === undefined) {
+		throw new Error(`Type rollup file '${typeRollupFilePath}' not found.`);
+	}
+	typeDefinitionFilePath = typeRollupFilePath;
+} else {
+	// Try to read the 'types' field from the package.json of the previous package
+	const previousPackageJson: PackageJson = readJsonSync(`${previousBasePath}/package.json`);
+	if (previousPackageJson.types) {
+		typeDefinitionFilePath = path.join(previousBasePath, previousPackageJson.types);
+	} else if (previousPackageJson.exports) {
+		// Function to extract types path from an export entry
+		const extractTypesPath = (exportEntry: any): string | undefined => {
+			return exportEntry?.types ? path.join(previousBasePath, exportEntry.types) : undefined;
+		};
+
+		const defaultSubpath = ".";
+		const exportEntry = previousPackageJson.exports[defaultSubpath];
+
+		// Check both 'import' and 'require' resolution methods
+		typeDefinitionFilePath =
+			extractTypesPath(exportEntry?.import) || extractTypesPath(exportEntry?.require);
```

Comment:
Preferable to use actual null/undefined coalescing. `||` could have subtly different meanings depending on context and/or actual runtime values.
```suggestion
		typeDefinitionFilePath =
			extractTypesPath(exportEntry?.import) ?? extractTypesPath(exportEntry?.require);
```